docker/archive/dest.go

```diff
@@ -19,17 +19,32 @@ func newImageDestination(ctx *types.SystemContext, ref archiveReference) (types.
 	if ref.destinationRef == nil {
 		return nil, errors.Errorf("docker-archive: destination reference not supplied (must be of form <path>:<reference:tag>)")
 	}
-	fh, err := os.OpenFile(ref.path, os.O_WRONLY|os.O_EXCL|os.O_CREATE, 0644)
+	// tries to open a file first to accomodate for load when using stdout
+	// as the file is created first then written to
+	// if the file doesn't exist then it creates a new file for exclusive use
+	// if the file exists, it checks to ensure the size is 0
```

Comment:
> We are trying to match the docker save command, their use case is to put multiple versions of the same image/tags into a tar ball at the same time.

Ah, OK. The containers/image design does make that awkward; there is no concept of copying a group of images at once, and appending images one by one would involve creating temporary tar file copies, potentially gigabytes in size—or, as discussed above, files with duplicate file names.

> Currently kpod load is able to read the images created with multiple tags

The containers/image archive reader currently refuses to process an archive with multiple images; it may support an image with multiple tags pointing at the same image, but only because the tag is ignored. (Though implementing the ability to read any of the images by tag should be quite possible.)

> It is curious that the documentation for docker load seems to indicate one image.

`docker load` does not allow you to _choose_ an image; it always import every image, however many there are, and it always sets all the tags set in the tarball, whatever they are. (I.e. loading a tarball from an untrusted source is not a good idea, it can redirect the `rhel:latest` link for example.)